src/openforms/payments/contrib/worldline/tests/test_migrations.py

```diff
@@ -0,0 +1,268 @@
+
+        self.assertEqual(merchants[2].label, "Merchant 3")
+        self.assertEqual(merchants[2].pspid, "merchant-3")
+        self.assertEqual(merchants[2].api_key, "merchant-3-api-key")
+        self.assertEqual(merchants[2].api_secret, "merchant-3-api-secret")
+        self.assertEqual(merchants[2].endpoint, WorldlineEndpoints.test)
+
+    def test_migrate_webhook_configuration(self):
+        WorldlineWebhookConfiguration = self.apps.get_model(
+            "payments_worldline", "WorldlineWebhookConfiguration"
+        )
+        webhook_configuration = WorldlineWebhookConfiguration.objects.get()
+
+        self.assertEqual(webhook_configuration.webhook_key_id, "webhook-key")
+        self.assertEqual(webhook_configuration.webhook_key_secret, "webhook-key-secret")
+
+
+class WorldlineDuplicateOgoneMigrationTest(TestMigrations):
+    migrate_from = "0006_worldlinewebhookconfiguration_and_more"
+    migrate_to = "0007_auto_20250829_1446"
+    app = "payments_worldline"
+    exception: Exception
+
+    def setUp(self):
+        try:
+            super().setUp()
+        except Exception as exc:
+            self.exception = exc
+
+    def setUpBeforeMigration(self, apps):
+        OgoneMerchant = apps.get_model("payments_ogone", "OgoneMerchant")
+        WorldlineMerchant = apps.get_model("payments_worldline", "WorldlineMerchant")
+
+        OgoneMerchant.objects.create(
+            label="Merchant 1",
+            pspid="merchant-1",
+            hash_algorithm=HashAlgorithm.sha512,
+            sha_in_passphrase="SHA-IN-PASSPHRASE",
+            sha_out_passphrase="SHA-OUT-PASSPHRASE",
+            api_key="merchant-1-api-key",
+            api_secret="merchant-1-api-secret",
+            endpoint_preset=OgoneEndpoints.live,
+        )
+        OgoneMerchant.objects.create(
+            label="Merchant 1 test",
+            pspid="merchant-1",
+            hash_algorithm=HashAlgorithm.sha512,
+            sha_in_passphrase="SHA-IN-PASSPHRASE",
+            sha_out_passphrase="SHA-OUT-PASSPHRASE",
+            api_key="merchant-1-test-api-key",
+            api_secret="merchant-1-test-api-secret",
+            endpoint_preset=OgoneEndpoints.test,
+        )
+        OgoneMerchant.objects.create(
+            label="Merchant 2",
+            pspid="merchant-2",
+            hash_algorithm=HashAlgorithm.sha512,
+            sha_in_passphrase="SHA-IN-PASSPHRASE",
+            sha_out_passphrase="SHA-OUT-PASSPHRASE",
+            api_key="merchant-2-api-key",
+            api_secret="merchant-2-api-secret",
+            endpoint_preset=OgoneEndpoints.live,
+        )
+        WorldlineMerchant.objects.create(
+            label="Merchant 2",
+            pspid="merchant-2",
+            api_key="merchant-2-api-key",
+            api_secret="merchant-2-api-secret",
+            endpoint=WorldlineEndpoints.live,
+        )
+
+    def test_duplicate_ogone_merchants(self):
+        executor = MigrationExecutor(connection)
```

Comment:
this seems too low-level, the whole purpose of `TestMigrations` is that you don't have to manually use these django internals.

This test also has very little added value, because the upgrade check with the check script that you're adding will block migrations from starting in the first place, exactly so that you don't have to worry about duplicates in the actual data migration.

This entire test case can be removed 😃